pkg/solver/controller.go

```diff
@@ -321,6 +322,16 @@ func (controller *SolverController) addResourceOffer(resourceOffer data.Resource
 	}
 	resourceOffer.ID = id
 
+	// Check the resource provider's ETH balance
+	balance, err := controller.web3SDK.GetBalance(resourceOffer.ResourceProvider)
+	if err != nil {
+		return nil, err
+	}
+	// If the balance is 0, don't add the resource offer
+	if balance.Cmp(big.NewInt(0)) == 0 {
+		return nil, err
```

Comment:
Can we log something here so we can track RPs that need to add eth? (or maybe this is a spot for some tracing, @bgins )?

Reply:
I think we can add tracking here as part of the tracing that @bgins is working on. We can put that on a separate PR.

Reply:
Yep, we could add tracing to report balances and resource providers that have been passed over for insufficient funds. 👌